### PR TITLE
fix: Remove THREE_SCALE_CONNECT_API and make API routes configurable

### PR DESCRIPTION
**Pull request by dvandra**
Configurable API routes:
`dependencyAnalytics.host`
`dependencyAnalytics.apiKey`
`dependencyAnalytics.stackReportUIHost`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -187,6 +187,18 @@
           "default": "off",
           "description": "Traces the communication between VSCode and the dependencyAnalyticsServer service."
         },
+        "dependencyAnalytics.host": {
+          "type": "string",
+          "default": "",
```
Comment (arajkumar) [inline package.json:192]: Should have default value which points to prod host.

File: package.json [hunk 2]
```diff
@@ -187,6 +187,18 @@
...
+          "description": "Specifies host api-route.",
+          "scope": "window"
+        },
+        "dependencyAnalytics.apiKey": {
+          "type": "string",
+          "default": "",
```
Comment (arajkumar) [inline package.json:198]: Should have default value which points to prod key.

File: src/Utils.ts [hunk 3]
```diff
@@ -3,6 +3,17 @@
...
 
 export namespace Utils {
+  export function getApiRoutes(): any {
+    let dependencyAnalytics = {
+      host: "",
+      apiKey: ""
+    }
+    const apiConfig = vscode.workspace.getConfiguration('dependencyAnalytics');
+    dependencyAnalytics.host = apiConfig.host ? `${apiConfig.host}` : 'https://f8a-analytics-2445582058137.production.gw.apicast.io';
+    dependencyAnalytics.apiKey = apiConfig.apiKey ? `${apiConfig.apiKey}` : '9e7da76708fe374d8c10fa752e72989f';
```
Comment (arajkumar) [inline src/Utils.ts:13]: This could be avoided if default value is specified in the config.

File: src/Utils.ts [hunk 4]
```diff
@@ -3,6 +3,17 @@
 import * as vscode from 'vscode';
```
Comment (arajkumar) [inline src/Utils.ts:3]: I would suggest to rename this module to "config" instead of "Util"

File: package.json [hunk 5]
```diff
@@ -187,6 +187,18 @@
...
+          "description": "Specifies host api-key.",
+          "scope": "window"
+        },
         "maven.executable.path": {
```
Comment (arajkumar) [inline package.json:202]: You must make stack report ui  host  also configurable.

**Review by arajkumar — CHANGES_REQUESTED** [review]

File: src/authextension.ts [hunk 6]
```diff
@@ -9,13 +9,13 @@ import { Utils } from './Utils';
...
   export let setContextData: any;
 
-  setContextData = (context_f8_access_routes, context_f8_3scale_user_key) => {
-    Apiendpoint.STACK_API_URL = Apiendpoint.STAGE_API_URL + '/api/v2/';
-    Apiendpoint.STACK_API_USER_KEY = Apiendpoint.STAGE_THREE_SCALE_USER_TOKEN;
-    Apiendpoint.OSIO_ROUTE_URL = Apiendpoint.STAGE_API_URL;
+  setContextData = (apiRoutes) => {
+    Apiendpoint.STACK_API_URL = apiRoutes.host + '/api/v2/';
+    Apiendpoint.STACK_API_USER_KEY = apiRoutes.apiKey;
+    Apiendpoint.OSIO_ROUTE_URL = apiRoutes.host;
```
Comment (prashbnair) [inline src/authextension.ts:15]: Are we still using the OSIO route?

File: src/apiendpoint.ts [hunk 7]
```diff
@@ -9,13 +9,5 @@ export namespace Apiendpoint {
   export let OSIO_ROUTE_URL: string = '';
   export const STACK_REPORT_URL: string =
```
Comment (prashbnair) [inline src/apiendpoint.ts:10]: Is this file required anymore? It just has a bunch of empty string declarations.

**Review by prashbnair — CHANGES_REQUESTED** [review]

File: src/authextension.ts (hunk 6 above)
Comment (arajkumar) [reply src/authextension.ts:15]: OSIO_ROUTE_URL points to same 3scale app host. +1 to remove OSIO_ROUTE_URL.

File: package.json [hunk 8]
```diff
@@ -187,6 +187,24 @@
           "default": "off",
           "description": "Traces the communication between VSCode and the dependencyAnalyticsServer service."
         },
+        "dependencyAnalytics.host": {
+          "type": "string",
+          "default": "https://f8a-analytics-2445582058137.production.gw.apicast.io",
+          "description": "Specifies host api-route.",
```
Comment (arajkumar) [inline package.json:193]: nit:
```suggestion
          "description": "Host name of Dependency Analytics server.",
```

File: package.json [hunk 9]
```diff
@@ -187,6 +187,24 @@
...
+          "scope": "window"
+        },
+        "dependencyAnalytics.apiKey": {
+          "type": "string",
+          "default": "9e7da76708fe374d8c10fa752e72989f",
+          "description": "Specifies host api-key.",
```
Comment (arajkumar) [inline package.json:199]: 3scale user_key to access dependencyAnalytics.host.

File: package.json [hunk 10]
```diff
@@ -187,6 +187,24 @@
...
+          "scope": "window"
+        },
+        "dependencyAnalytics.stackReportUIHost": {
+          "type": "string",
+          "default": "https://stack-analytics-report.openshift.io/",
+          "description": "Specifies host for Stack Report UI",
```
Comment (arajkumar) [inline package.json:205]:
```suggestion
          "description": "Host name of the stack report UI.",
```

File: src/Config.ts [hunk 11]
```diff
@@ -2,7 +2,12 @@
 
 import * as vscode from 'vscode';
```
Comment (arajkumar) [inline src/Config.ts:4]: File name should start with lower case.

rename this to config.ts

File: src/Config.ts [hunk 12]
```diff
@@ -2,7 +2,12 @@
...
 
-export namespace Utils {
+export namespace Config {
+
+  export const apiConfig = vscode.workspace.getConfiguration('dependencyAnalytics');
```
Comment (arajkumar) [inline src/Config.ts:7]: You should call this every time, otherwise the new value won't get updated.

File: src/stackanalysismodule.ts [hunk 13]
```diff
@@ -128,7 +128,7 @@ export module stackanalysismodule {
     uri = null
   ) => {
     let effectiveF8Var: string, argumentList: string;
-    Apiendpoint.API_ECOSYSTEM = ecosystem;
+    Config.API_ECOSYSTEM = ecosystem;
```
Comment (arajkumar) [inline src/stackanalysismodule.ts:131]: What is the point of having this global var? This could be removed by passing ecosystem to stackanalysismodule.stackAnalysesLifeCycle

File: src/multimanifestmodule.ts [hunk 14]
```diff
@@ -8,7 +8,8 @@ import { stackanalysismodule } from './stackanalysismodule';
 import { authextension } from './authextension';
 import { StatusMessages } from './statusMessages';
 import { DependencyReportPanel } from './dependencyReportPanel';
-import { Apiendpoint } from './apiendpoint';
+import { Config } from './Config';
+import { config } from 'process';
```
Comment (mathur07) [inline src/multimanifestmodule.ts:12]: @dvandra can you take a look, is this import required?

**Review by mathur07 — CHANGES_REQUESTED** [review]

**Review by arajkumar — CHANGES_REQUESTED** [review]
few nits, overall changes looks okay to me.

**Review by dgpatelgit — APPROVED** [review]
lgtm

File: src/multimanifestmodule.ts (hunk 14 above)
Comment (dvandra) [reply src/multimanifestmodule.ts:12]: No, I missed this to remove.

**Review by prashbnair — APPROVED** [review]

File: src/stackanalysismodule.ts (hunk 13 above)
Comment (arajkumar) [reply src/stackanalysismodule.ts:131]: I see that ecosystem stored in global variable to pass around. This shall be solved by argumenting it to downstream functions.

**Review by mathur07 — APPROVED** [review]
LGTM

**Comment by mathur07** [comment]
[test]

**Comment by arajkumar** [comment]
Just started reviewing :)

**Review by arajkumar — APPROVED** [review]
lgtm
